### ISSUE TITLE
Request.Options.writeTimeoutMillis

### ISSUE THREAD

**Issue by uschi2000** [Netflix/feign#340]
Hi all,
Is there a reason for `Request.Options` not to offer a writeTimeoutMillis option?
  Thanks! Robert.

````java
    public Options(int connectTimeoutMillis, int readTimeoutMillis) {
      this.connectTimeoutMillis = connectTimeoutMillis;
      this.readTimeoutMillis = readTimeoutMillis;
    }
````

**Comment by adriancole** [comment]
probably because of the context of writes in Feign's design (buffered into byte arrays), this hasn't come up. Ex uploads were always quite small compared to downloads. I presume you've run into a write timeout? :) To consider this, we'd have to look at the clients to see if they can support it. Do you have time to audit this?

**Comment by uschi2000** [comment]
Yeah, we have use-cases with long-lived writes. I have only checked okhttp so far, but can take a look at the other ones too.

**Comment by kdavisk6** [comment]
@uschi2000 

If you are still interested in pursuing this feature, please let me know.  If not, please close this issue.
